File: libpandadna/DNABattleCell.py
from common import *

class DNABattleCell:
    COMPONENT_CODE = 21

    def __init__(self, width, height, pos):
        self.width = width
        self.height = height
        self.pos = pos

    def setWidth(self, width):
        self.width = width

    def setHeight(self, height):
        self.height = height

    def setWidthHeight(self, width, height):
        self.setWidth(width)
        self.setHeight(height)

    def setPos(self, pos):
        self.pos = pos

    def getWidth(self):
        return self.width
        
    def getHeight(self):
        return self.height
        
    def getPos(self):
        return Point3(self.pos)
        

File: src/components/DNABattleCell.cxx
#include "DNABattleCell.h"

TypeHandle DNABattleCell::_type_handle;

DNABattleCell::DNABattleCell(float width, float height,
                             LPoint3f pos): m_width(width),
                             m_height(height), m_pos(pos)
{
}

DNABattleCell::~DNABattleCell()
{
}
